node/coinstacks/bitcoin/api/src/controller.ts

```diff
@@ -138,62 +141,138 @@ export class Bitcoin extends Controller implements BaseAPI, BitcoinAPI {
         blockHeight: 645850,
         timestamp: 1598700231,
+        confirmations: 81971,
+        value: '510611',
+        fee: '224',
+        hex:
+          '020000000158c53ac5b9e1b56b571c9530dc214af7be2efae45974a337d24e555d77d57b16010000006a47304402200e13f2e48612bfe1b8d69f59dfcfc288f3f896a248aac7d70b1dcc2e860e2f70022057487d594e247d6980b330e767c9e2b3e99dc975481999bb0307b5dc96c3923b012102174a52f766c3c174adbcaf1677818e04879cb8f27153d7a44775b050e11a44e0feffffff02090300000000000017a9140f7f0fc4f882ea62f32b06f0946f12b055ab91bf878ac70700000000001976a9148c1ef82c52a7e80621c838008b2de791be3a307988ac98d80900',
+        vin: [
+          {
+            txid: '167bd5775d554ed237a37459e4fa2ebef74a21dc30951c576bb5e1b9c53ac558',
+            vout: '1',
+            sequence: 4294967294,
+            scriptSig: {
+              hex:
+                '47304402200e13f2e48612bfe1b8d69f59dfcfc288f3f896a248aac7d70b1dcc2e860e2f70022057487d594e247d6980b330e767c9e2b3e99dc975481999bb0307b5dc96c3923b012102174a52f766c3c174adbcaf1677818e04879cb8f27153d7a44775b050e11a44e0',
+            },
+            addresses: ['1Dmthegfep7fXVqWAPmQ5rMmKcg58GjEF1'],
+          },
+        ],
+        vout: [
+          {
+            value: '777',
+            n: 0,
+            scriptPubKey: {
+              hex: 'a9140f7f0fc4f882ea62f32b06f0946f12b055ab91bf87',
+              addresses: ['336xGpGweq1wtY4kRTuA4w6d7yDkBU9czU'],
+            },
+          },
+          {
+            value: '509834',
+            n: 1,
+            scriptPubKey: {
+              hex: '76a9148c1ef82c52a7e80621c838008b2de791be3a307988ac',
+              addresses: ['1Dmthegfep7fXVqWAPmQ5rMmKcg58GjEF1'],
+            },
+          },
+        ],
       },
     ],
   })
   @Response<BadRequestError>(400, 'Bad Request')
   @Response<ValidationError>(422, 'Validation Error')
   @Response<InternalServerError>(500, 'Internal Server Error')
   @Get('account/{pubkey}/txs')
-  async getTxHistory(@Path() pubkey: string, @Query() cursor?: string, @Query() pageSize = 25): Promise<TxHistory> {
+  async getTxHistory(
+    @Path() pubkey: string,
+    @Query() cursor?: string,
+    @Query() pageSize = 10
+  ): Promise<BitcoinTxHistory> {
     try {
-      let page = 1
-      if (cursor) {
-        page = Number(Buffer.from(cursor, 'base64').toString('binary'))
-        if (isNaN(page) || page < 1) {
-          throw new ApiError('Bad Request', 400, 'Invalid cursor, please use the cursor returned in previous page')
+      const curCursor = ((): Cursor => {
+        try {
+          if (!cursor) return { page: 1 }
+
+          const decodedCursor = JSON.parse(Buffer.from(cursor, 'base64').toString('binary'))
+
+          // Validate that the cursor contains a 'page' property this is a positive integer
+          if (!('page' in decodedCursor && Number.isInteger(decodedCursor.page) && decodedCursor.page >= 1)) {
+            throw 'invalid base64 cursor'
```

Comment:
Would be useful to throw this error with the parsed cursor just for extra insight. `throw new Error(`invalid json cursor: ${JSON.stringify(decodedCursor)`)`

I also did this check outside of the decode try/catch so we can throw a more specific error (invalid base64 cursor vs invalide json cursor):
```
    if (curCursor.page === undefined) {
      const e: BadRequestError = { error: `invalid json cursor: ${JSON.stringify(curCursor)}` }
      throw new ApiError('Bad Request', 400, JSON.stringify(e))
    }
```